war/src/main/js/add-item.js

```diff
@@ -0,0 +1,414 @@
+// Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
+var $jq = require('jquery-detached').getJQuery();
+var jenkinsLocalStorage = require('./util/jenkinsLocalStorage.js');
+var itemIcons = {
+    'hudson.model.FreeStyleProject':'freestyle-48.png',
+    'hudson.maven.MavenModuleSet':'maven-48.png',
+    'org.jenkinsci.plugins.workflow.job.WorkflowJob':'pipeline-48.png',
+    'com.infradna.hudson.plugins.backup.BackupProject':'backup-48.png',
+    'hudson.model.ExternalJob':'remote-job-48.png',
+    'com.cloudbees.jenkins.plugins.longrunning.LongRunningProject':'long-running-48.png',
+    'hudson.matrix.MatrixProject':'multi-config-48.png',
+    'com.cloudbees.hudson.plugins.folder.Folder':'folder-48.png',
+    'org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject':'branch-project-48.png',
+    'jenkins.branch.OrganizationFolder.org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator':'gitHub-project-48.png',
+    'jenkins.branch.OrganizationFolder.com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMNavigator':'bitBucket-project-48.png',
+    'copy':'copy-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.jobTemplate.JobTemplate':'job-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.folder.FolderTemplate':'folder-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.auxiliary.AuxModel':'aux-template-48.png', 
+    'com.cloudbees.hudson.plugins.modeling.impl.builder.BuilderTemplate':'builder-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.publisher.PublisherTemplate':'publish-template-48.png'
+} 
```

Comment:
So this is still all fiction at the moment. How these items get connected to the to the items types is still way up in the air. This was most expedient for demonstration purposes.